zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorage.java

```diff
@@ -66,7 +66,12 @@
    */
   public FileSystemBackupStorage(BackupConfig backupConfig) {
     this.backupConfig = backupConfig;
-    fileRootPath = String.join(File.separator, this.backupConfig.getBackupStoragePath(),
+    String backupStoragePath = this.backupConfig.getBackupStoragePath();
+    if (!new File(backupStoragePath).exists()) {
+      throw new BackupException(
+          "The backup storage is not ready, please check the path: " + backupStoragePath);
+    }
```

Comment:
Let's do this check right in the beginning?

Reply:
Synced offline with @narendly. 
If storagePath is not set, it will return "" so an NPE won't happen.
Moved the check to the beginning.